crypto/keys/src/ristretto.rs

```diff
@@ -105,7 +105,19 @@ impl PrivateKey for RistrettoPrivate {
 
 impl FromRandom for RistrettoPrivate {
     fn from_random<R: CryptoRng + RngCore>(csprng: &mut R) -> RistrettoPrivate {
-        Self(Scalar::random(csprng))
+        let mini_secret = MiniSecretKey::generate_with(csprng);
+        let secret = mini_secret
+            .expand_to_keypair(ExpansionMode::Uniform)
+            .secret
+            .clone();
+        // The scalar is the first 32 bytes in canonical representation
+        let mut scalar_bytes: [u8; 32] = [0u8; 32];
+        scalar_bytes[0..32].copy_from_slice(&secret.to_bytes()[0..32]);
+        // FIXME: error handling
```

Comment:
fwiw I think the code as written is correct, there should not be any error case for generating a ristretto private from an rng, and if we want to get rid of these .expect then we should fix schnorrkel crate and not attempt to bubble these errors up.

Reply:
generating a random private key cannot logically fail. if you feel tempted to bubble errors out of that then something is wrong

Reply:
Ok great - I will leave as `expect()`